### PR TITLE
Addition to TASTy reflect to semantic DB

### PR DESCRIPTION
**Pull request by poechsel**
PR from my work on a tasty to semanticDB converter.

This PR is mostly to merge the addition I've made to tasty-reflect (added an accessor for `exists` on `Position` and accessors to `companionClass`, `companionModule` and `pos` on `Symbol`).

The converter itself is under ongoing work and is not production ready.

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/tastyreflect/SymbolOpsImpl.scala [hunk 1]
```diff
@@ -26,8 +26,14 @@ trait SymbolOpsImpl extends scala.tasty.reflect.SymbolOps with CoreImpl {
     def name(implicit ctx: Context): String = symbol.name.toString
     def fullName(implicit ctx: Context): String = symbol.fullName.toString
 
+    def pos(implicit ctx: Context): Position = symbol.pos
+
     def owner(implicit ctx: Context): Symbol = symbol.owner
 
+    def companionModule(implicit ctx: Context): Symbol = symbol.companionModule
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/tastyreflect/SymbolOpsImpl.scala]: Now they are in `ClassSymbol`

File: semanticdb/Makefile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+path="input/src/main/scala/"
+files=$(shell find $(path) -type f -name '*.scala')
```
Comment (nicolasstucki) [inline semanticdb/Makefile]: Add README

**Comment by nicolasstucki** [comment]
@poechsel looks like you have to rebase. There are some conflicts with documentation lines.

File: semanticdb/Makefile [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+all: %.tasty
```
Comment (smarter) [inline semanticdb/Makefile]: I'd rather not add another build system to our project. Can this be replaced by one or more sbt tasks ?

Comment (nicolasstucki) [reply semanticdb/Makefile]: It should be possible

Comment (poechsel) [reply semanticdb/Makefile]: I removed the Makefile. I didn't saw that the `*.tasty` files I needed were already generated by `sbt`. I'm now using them directly. However, to run the tests it is still needed to run `sbt compile` inside `semanticdb/input` as described in #5253.

Comment (smarter) [reply semanticdb/Makefile]: semanticdb/input/build.sbt should be merged into our project/Build.scala to remove the need for this.

File: project/Build.scala [hunk 4]
```diff
@@ -1317,6 +1322,7 @@ object Build {
       enablePlugins(JmhPlugin)
 
     def asDottySemanticdb(implicit mode: Mode): Project = project.withCommonSettings.
+      aggregate(`dotty-semanticdb-input`).
```
Comment (nicolasstucki) [inline project/Build.scala]: Can't you just depend on it?

**Pull request merged**